app/main.py

```diff
@@ -1,2 +1,48 @@
+from typing import Hashable, Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length = 0
+        self.hash_table: list = [None] * 8
+        self.initial_capacity = 8
+        self.load_factor = self.initial_capacity * 2 // 3
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        self.length += 1
+        if self.length > self.load_factor:
+            self.resize()
+
+        self.hash_table[self.get_index(key)] = (key, hash(key), value)
+
+    def __getitem__(self, key: Hashable) -> Any:
+        for item in self.hash_table:
+            if item and key == item[0]:
+                return item[2]
+        raise KeyError
+
+    def __len__(self) -> int:
+        return self.length
+
+    def resize(self) -> None:
+        new_hash_table: list = [None] * (self.initial_capacity * 2)
+        for i in range(self.initial_capacity):
+            if self.hash_table[i]:
+                new_hash_table[i] = self.hash_table[i]
```

Comment:
after changing the size of the list, the indexes on which the elements should appear may differ from those that were previously. For rxample key == 9, hash(9) =9 , capacity == 8, index should be 9 % 8 == 1, but if capacity  == 16  index should be 9 % 16 == 9